docs/source/monitors.rst

```diff
@@ -32,8 +32,10 @@ number of items were extracted and fails if it is less than the expected thresho
                 msg='Extracted less than {} items'.format(minimum_threshold)
             )
 
-A `Monitor` instance has the following properties that can be used to help you
-implement your monitors:
+.. class:: Monitor(name=name, data=None, parent=None)
```

Comment:
The reference documentation should be at the end of this section, not in the middle.

Also, you should use `.. autoclass:: spidermon.core.monitors.Monitor`, followed by `:members:` in the following line, to get the documentation automatically loaded from the source code (when available).

Reply:
Roger that, it's my time working with class references in Sphinx. Good to know stuff!

Reply:
Where exactly do we need this section to be at, I feel this should be referenced when we actually define what are monitors.

Reply:
API reference usually goes at the end. First you introduce the concepts, basic usage, special scenarios, and at the end of it all goes the API reference, for people who still have questions after reading everything else about the class.

If you look at the section about the `MonitorSuite` class, it is also at the end of its section.

Reply:
Roger that, followed your advice. Let me know if this location is right, if not please let me know the exact location you would need it to be. It's just that I have a hard time figuring out where to place it best since the entire page is about monitors and placing it at the very end is weird. 

Also, should we add some documentation for the class `spidermon.core.monitors.Monitor` in the code? Seeing it empty looks a bit unusual right now. What say?

Reply:
I did not meant at the end of the page, but at the end of the section. In other words, right above the Monitor Suites header.

As for documenting `spidermon.core.monitors.Monitor` in code, it definitely needs to be done eventually. It’s up to you whether you want to make it part of this changeset or leave it for a later changeset.

Reply:
@Gallaecio No worries but isn't that the same place I put the class tag at the very start :sweat_smile: Since the documentation for `spidermon.core.monitors.Monitor` is not available. I thought, keeping it above where we define the `Monitor` instance would be more appropriate. I know it doesn't look good right now but sure is better than having some explanation than no explanation. Let me know how you would like it. 
  
![Screenshot_2019-10-23_10-53-26](https://user-images.githubusercontent.com/22801822/67360396-712ff500-f583-11e9-8332-dce1735e4cfb.png)

I can surely add the docs for that in the code later, as part of another PR. WIll open an issue as well, if someone from the Hacktoberfest picks it up. Could lead to potential new contributors.

Reply:
I believe in your changes when I originally commented it was placed before the “A Monitor instance…” paragraph: https://github.com/scrapinghub/spidermon/commit/c113537d395421aba33c8e2ad9b7280fca032ab9

Reply:
+1 then, resolving this!